docs/topics/class-based-views/advanced.txt

```diff
@@ -0,0 +1,694 @@
+===========================================
+Advanced customization of class-based views
+===========================================
+
+.. caution::
+
+    This is an advanced topic. A working knowledge of :doc:`Django's
+    class-based views<index>` is advised before exploring these
+    techniques.
+
+This section is meant to provide the following:
+
+* a number of ready-to-use customized class-based view examples
+
+* a walk-through of the process of creation and design of provided examples
+
+* a discussion of pro's and con's of different approaches when customizing
+  class-based views
+
+* a specific focus on a few areas of class-based generic view hierarchy that
+  can be complex and / or daunting to tweak
+
+There are two possible uses for customized class-based generic views: they may
+serve as parent classes for future inheritance, or they may be used as "Leaf
+Classes" to serve actual views in your application.
+
+In the former case, the general approach should be to override all of the
+methods where functionality needs to be added by the custom class. The
+resulting class should be as consistent with existing generic hierarchy as
+possible. This allows for maximum flexibility in future reuse and further
+customization by the "Leaf Classes".
+
+In the latter case, when using a class as actual view, you may find it useful
+to use an almost exactly opposite approach:
+
+  * flatten and simplify the hierarchy when possible
+
+  * reduce the number of methods called when class is used
+
+  * prefer to methods directly instead of indirectly through intermediary methods
+
+The idea being that the operation of the view should be explicit and easy to
+follow, to debug and modify (albeit at the cost of allowing less flexibility
+for future reuse through inheritance).
+
+In this section, we will mostly focus on the latter approach -- customizing
+classes to be used as actual views.
+
+Combined detail / list view
+===========================
```

Comment:
In this view, what does `model` refer to? It's the detail, but this isn't immediately clear. `get_queryset()`, which would normally be your start point for customising the list part, will actually customise the base queryset for `self.object`